### ISSUE TITLE
System.PlatformNotSupportedException when using ContentDisposition 

### ISSUE THREAD

**Issue by pureggerstefan** [dotnet/runtime#42576]
Greetings, 

since the .NET5 RC1 update on a Blazor WASM, following lines lead to a System.PlatformNotSupportedException

```c#
 private static string GetFilename(IReadOnlyDictionary<string, IEnumerable<string>> headers)
        {
            if (headers.ContainsKey("Content-Disposition"))
            {
                //Exception throws at:
                var contentDisposition = new ContentDisposition(headers["Content-Disposition"].First());

                return contentDisposition.FileName;
            }

            return string.Empty;
        }
```

I think this is a bug, since the ContentDisposition class documentation says it applies to .NET RC1.

Best regards,
Stefan

**Comment by stephentoub** [comment]
cc: @akoeplinger

**Comment by MaximLipnin** [comment]
JFR - `System.Net.Mail` has been modified to throw PNSE at assembly level on Browser (see https://github.com/dotnet/runtime/pull/38207)

**Comment by marek-safar** [comment]
Yes, we will probably have to exclude ContentDisposition from that if it's also used by System.Net.Http headers

**Comment by LaughingJohn** [comment]
I've just stumbled over this as well. Different scenario - I was using MailAddress in Blazor webassembly to validate an email address and it stopped working when I converted it to .Net 5.0. It used to work with .Net core.

`
            try
            {
                new MailAddress(emailAdddress);
            }
            catch
            {
                return false;
            }
`

**Comment by kjpou1** [comment]
Looking into this

**Comment by kjpou1** [comment]
Closed by https://github.com/dotnet/runtime/pull/42974

**Comment by karelz** [comment]
Fixed in 6.0/master (PR #42974) and in 5.0 (PR #43213).
